raft/raft.go

```diff
@@ -346,7 +363,18 @@ func (r *raft) hardState() pb.HardState {
 	}
 }
 
-func (r *raft) quorum() int { return len(r.prs)/2 + 1 }
+// TODO(shuaili): optimize, maybe record voterCount in raft
+func (r *raft) voterCount() int {
+	count := 0
+	for _, p := range r.prs {
+		if !p.isLearner {
+			count++
+		}
+	}
+	return count
+}
+
+func (r *raft) quorum() int { return r.voterCount()/2 + 1 }
```

Comment:
This function is called frequently, maybe now we can use a variable to cache the quorum.

Reply:
@xiang90

Reply:
agreed